pkg/scheduler/core/util.go

```diff
@@ -68,22 +72,51 @@ func calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha
 		return availableTargetClusters
 	}
 
-	// Get the minimum value of MaxAvailableReplicas in terms of all estimators.
-	estimators := estimatorclient.GetReplicaEstimators()
+	// Get all replicaEstimators, which are stored in TreeMap.
+	replicaEstimators := estimatorclient.GetReplicaEstimators()
 	ctx := context.WithValue(context.TODO(), util.ContextKeyObject,
 		fmt.Sprintf("kind=%s, name=%s/%s", spec.Resource.Kind, spec.Resource.Namespace, spec.Resource.Name))
-	for _, estimator := range estimators {
-		res, err := estimator.MaxAvailableReplicas(ctx, clusters, spec.ReplicaRequirements)
-		if err != nil {
-			klog.Errorf("Max cluster available replicas error: %v", err)
-			continue
+
+	// List all replicaEstimators in order of descending priority. The estimators are grouped with different priorities,
+	// e.g: [priority:20, {estimators:[es1, es3]}, {priority:10, estimators:[es2, es4]}, ...]
+	estimatorGroups := replicaEstimators.Values()
+
+	// Iterate the estimator groups in order of descending priority
```

Comment:
From the perspective of code implementation, this is just a simple multi-level loop, and I don't think grouping has any advantages but increases code complexity (two-level loops become three-level loops, although the total number of loops remains unchanged). So the main reason is that I don't see how the advantages of grouping are used.

Reply:
@Garrybest @RainbowMango This is just what I memtioned above that `Complex structures or more loops can also complicate the problem.`

This is consistent with my initial view, but since it has been changed, I don't want to change it back. What do you all think?